packages/@aws-cdk/aws-scheduler-alpha/lib/private/schedule.ts

```diff
@@ -36,19 +37,31 @@ export interface ScheduleProps {
    * @default - no value
    */
   readonly description?: string;
+
+  /**
+   * The schedule's group.
+   *
+   * @deafult - By default a schedule will be associated with the `default` group.
+   */
+  readonly group?: IGroup;
 }
 
 /**
  * An EventBridge Schedule
  */
 export class Schedule extends Resource implements ISchedule {
+  group?: IGroup;
+
   constructor(scope: Construct, id: string, props: ScheduleProps) {
     super(scope, id);
 
+    this.group = props.group;
+
     new CfnSchedule(this, 'Resource', {
       flexibleTimeWindow: { mode: 'OFF' },
       scheduleExpression: props.schedule.expressionString,
       scheduleExpressionTimezone: props.schedule.timeZone?.timezoneName,
+      groupName: Lazy.string({ produce: () => this.group?.groupName }),
```

Comment:
Why does this one need to be lazy?

Reply:
> Why does this one need to be lazy? 

Because group can be assigned dynamically later in the code after schedule is created. 

```
const schedule1  = new Schedule(...);
const group = new Group(...);

group.addSchedules(schedule1);
```

Reply:
Good point :-) Totally missed that.